packages/components/src/navigator/navigator-link/component.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	WordPressComponentProps,
+} from '../../ui/context';
+import { View } from '../../view';
+import useNavigator from '../use-navigator';
+import type { NavigatorLinkProps } from '../types';
+
+const defaultAttributeName = 'id';
+const defaultSelectorFactory = ( attrName: string, attrValue: string ) =>
+	`[${ attrName }="${ attrValue }"]`;
+
+function NavigatorLink(
+	props: WordPressComponentProps< NavigatorLinkProps, 'a' >,
+	forwardedRef: Ref< any >
+) {
+	const {
+		path,
+		children,
+		onClick,
+		as = 'button',
```

Comment:
I guess same thing here, would be cool to reuse Button

Reply:
I can definitely look into changing it to `Button`.

For this component specifically, do you think that using `href` instead of `path` would be a better idea?

In my opinion, I like the idea that the actual rendered DOM for this component would be an `<a />` rather than a `<button />` — it feels better semantically (although those "paths" wouldn't be correct at a URL).

(for completeness: I believe that `Button`, when passed an `href` prop, automatically renders as a `<a />` instead of a `<button />`)

Reply:
> In my opinion, I like the idea that the actual rendered DOM for this component would be an <a /> rather than a <button /> — it feels better semantically (although those "paths" wouldn't be correct at a URL).

I'm not sure :) I think from an a11y perspective if it doesn't change the URL it should be a button but I may be wrong here. Semantically, we're not navigating to another page, we're opening a "view" or something like that I guess.

Reply:
I'm also not the best person to reply here :)

Reply:
I refactored to use `Button` by default in https://github.com/WordPress/gutenberg/pull/37223/commits/d219320891d6554e5a09b422a3c171cea5158412

> I'm also not the best person to reply here :)

Your feedback has already been very valuable. I guess I'll ask other folks for feedback later, once this PR reaches a more stable place